packages/mobile/src/components/track-list/TrackList.tsx

```diff
@@ -0,0 +1,116 @@
+import { ID } from 'audius-client/src/common/models/Identifiers'
+import { CoverArtSizes } from 'audius-client/src/common/models/ImageSizes'
+import { User } from 'audius-client/src/common/models/User'
+import { FlatList, View } from 'react-native'
+
+import { makeStyles } from 'app/styles'
+
+import { TrackItemAction, TrackListItem } from './TrackListItem'
+
+type TrackListProps = {
+  containerClassName?: string
+  itemClassName?: string
+  tracks: Array<{
+    isLoading: boolean
+    isSaved?: boolean
+    isReposted?: boolean
+    isActive?: boolean
+    isPlaying?: boolean
+    isRemoveActive?: boolean
+    artistHandle: string
+    artistName: string
+    trackTitle: string
+    trackId: ID
+    uid?: string
+    time?: number
+    coverArtSizes?: CoverArtSizes
+    isDeleted: boolean
+    user: User
+  }>
+  showTopDivider?: boolean
+  showDivider?: boolean
+  noDividerMargin?: boolean
+  showBorder?: boolean
+  onSave?: (isSaved: boolean, trackId: ID) => void
+  onRemove?: (index: number) => void
+  togglePlay?: (uid: string, trackId: ID) => void
+  trackItemAction?: TrackItemAction
+  isReorderable?: boolean
+  onReorder?: (index1: number, index2: number) => void
+}
+
+const useStyles = makeStyles(({ palette, spacing }) => ({
+  divider: {
+    borderBottomColor: palette.neutralLight7,
+    borderBottomWidth: 1,
+    marginVertical: 0,
+    marginHorizontal: spacing(6)
+  },
+  noMarginDivider: {
+    borderBottomColor: palette.neutralLight8,
+    marginHorizontal: 0
+  },
+  hideDivider: {
+    opacity: 0
+  }
+}))
+
+export const TrackList = ({
+  noDividerMargin,
+  onSave,
+  showDivider,
+  showTopDivider,
+  togglePlay,
+  trackItemAction,
+  tracks
+}: TrackListProps) => {
+  const styles = useStyles()
+
+  // The dividers above and belove the active track should be hidden
+  const activeIndex = tracks.findIndex(track => track.isActive)
+  const hideDivider = (idx: number) =>
+    activeIndex >= 0 && (activeIndex === idx || activeIndex === idx - 1)
+
+  const renderTrack = ({ item: track, index: idx }) => (
+    <View key={track.uid}>
+      {showDivider && (showTopDivider || idx > 0) ? (
+        <View
+          style={[
+            styles.divider,
+            hideDivider(idx) ? styles.hideDivider : {},
```

Comment:
hideDivider(idx) && styles.hideDivider also works. if we want to do ternery like we do with rendering, i think can just return null for else case, instead of empty object